micros.asd

```diff
@@ -1,41 +1,48 @@
 (defsystem "micros"
-  :depends-on ()
+  :depends-on (:bordeaux-threads)
```

Comment:
Adding a dependency in micros may risk a conflict between the library used by the user and the library used by micros, causing a malfunction.

Is there any way to avoid this?

Reply:
Ummm, I don't know really, if we want to make this compiled alone (without the Lem system),

Reply:
For example, if users are developing bordeaux-threads, changing their implementation will likely cause micros to stop working.

Reply:
oooh, I see ummm, then maybe use some specific functions depending on the implementation, a little more manual, but that can be a solution

Reply:
Ok, so.... SBCL implement a couple of more things for hash-tables to work concurrently so I'm experimenting with https://github.com/no-defun-allowed/concurrent-hash-tables .

It doesn't solve the problem, given that we still have to use external dependencies, so I think if this version works for more implementation, the solution might be to fork it and integrate it in micro as a internal system.

Reply:
It's a pain if there is a but, because we would have to go to the main repo to check it, buut I'll add the default option for default SBCL to work with the native calls, so at least it works as intended in the reference implementation.

Reply:
The concurrent-hash-table is very interesting, thanks for sharing.

In this case, we need a mutex of hash-table in eval-for-emacs, but I don't think it requires much performance.
What about going with a simple with-lock-held pattern?

Reply:
Still, I would have to use the bt library for that

Reply:
micros(swank) has definterface and defimplementation.
Since we already have the bare minimum for threads, why not just use them?

Reply:
So, seems like swank already have some thread-locks implementations :facepalm: , the code seems fine but it doesn't seem to work for SBCL.